pkg/resources/standard_database.go

```diff
@@ -0,0 +1,714 @@
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			databaseSetRequest.ReplaceInvalidCharacters = sdk.Bool(*replaceInvalidCharacters)
+		} else {
+			databaseUnsetRequest.ReplaceInvalidCharacters = sdk.Bool(true)
+		}
+	}
+
+	if d.HasChange("default_ddl_collation") {
+		defaultDdlCollationObject, ok := d.GetOk("default_ddl_collation")
+		if ok && len(defaultDdlCollationObject.([]any)) > 0 {
+			defaultDdlCollation, err := GetPropertyOfFirstNestedObjectByValueKey[string](d, "default_ddl_collation")
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			databaseSetRequest.DefaultDDLCollation = defaultDdlCollation
+		} else {
+			databaseUnsetRequest.DefaultDDLCollation = sdk.Bool(true)
+		}
+	}
+
+	if d.HasChange("storage_serialization_policy") {
+		storageSerializationPolicyObject, ok := d.GetOk("storage_serialization_policy")
+		if ok && len(storageSerializationPolicyObject.([]any)) > 0 {
+			storageSerializationPolicy, err := GetPropertyOfFirstNestedObjectByValueKey[string](d, "storage_serialization_policy")
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			databaseSetRequest.StorageSerializationPolicy = sdk.Pointer(sdk.StorageSerializationPolicy(*storageSerializationPolicy))
+		} else {
+			databaseUnsetRequest.StorageSerializationPolicy = sdk.Bool(true)
+		}
+	}
+
+	if d.HasChange("log_level") {
+		logLevelObject, ok := d.GetOk("log_level")
+		if ok && len(logLevelObject.([]any)) > 0 {
+			logLevel, err := GetPropertyOfFirstNestedObjectByValueKey[string](d, "log_level")
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			databaseSetRequest.LogLevel = sdk.Pointer(sdk.LogLevel(*logLevel))
+		} else {
+			databaseUnsetRequest.LogLevel = sdk.Bool(true)
+		}
+	}
+
+	if d.HasChange("trace_level") {
+		traceLevelObject, ok := d.GetOk("trace_level")
+		if ok && len(traceLevelObject.([]any)) > 0 {
+			traceLevel, err := GetPropertyOfFirstNestedObjectByValueKey[string](d, "trace_level")
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			databaseSetRequest.TraceLevel = sdk.Pointer(sdk.TraceLevel(*traceLevel))
+		} else {
+			databaseUnsetRequest.TraceLevel = sdk.Bool(true)
+		}
+	}
+
+	if d.HasChange("replication") {
+		before, after := d.GetChange("replication")
+
+		var (
+			accountsToEnableReplication  []sdk.AccountIdentifier
+			accountsToDisableReplication []sdk.AccountIdentifier
+			accountsToEnableFailover     []sdk.AccountIdentifier
+			accountsToDisableFailover    []sdk.AccountIdentifier
+
+			// maps represent replication configuration by having sdk.AccountIdentifier
+			// as a key (implicitly enabling replication), and failover as an option value
+			beforeReplicationFailoverConfigurationMap = make(map[sdk.AccountIdentifier]bool)
+			afterReplicationFailoverConfigurationMap  = make(map[sdk.AccountIdentifier]bool)
+		)
+
+		fillReplicationMap := func(replicationConfigs []any, replicationFailoverMap map[sdk.AccountIdentifier]bool) {
```

Comment:
Yeah, but I think that's a slightly different case than others, because it's a list of objects which is not trivial to compare and return only changed parameters in those nested objects to create 4 separate lists of added removed things. I think I would leave such cases, but for simple lists/sets of strings we could create such helper.